src/tools/fdsdump/src/lister/jsonPrinter.cpp

```diff
@@ -52,19 +52,41 @@ JsonPrinter::parse_fields(const std::string &str, const shared_iemgr &iemgr)
 void
 JsonPrinter::parse_opts(const std::string &str)
 {
-    std::vector<std::string> opt_names;
+    std::vector<std::string> opts;
 
     if (str.empty()) {
         return;
     }
 
-    opt_names = string_split(str, ",");
+    opts = string_split(str, ",");
 
-    for (const std::string &opt_raw : opt_names) {
+    for (const std::string &opt_raw : opts) {
         std::string opt = string_trim_copy(opt_raw);
+        std::vector<std::string> opt_parts = string_split(opt, "=", 2);
+        const std::string &opt_name = (opt_parts.size() >= 1) ? opt_parts[0] : "";
+        const std::string &opt_value = (opt_parts.size() >= 2) ? opt_parts[1] : "";
 
         if (strcasecmp(opt.c_str(), "no-biflow-split") == 0) {
             m_biflow_split = false;
+
+        } else if (strcasecmp(opt_name.c_str(), "timestamp") == 0) {
+            if (opt_value.empty()) {
+                throw std::invalid_argument(
+                    "JSON output: timestamp option is missing value"
+                    ", use 'timestamp=unix' or 'timestamp=formatted'");
+            }
+
+            const std::string &opt_value = opt_parts[1];
```

Comment:
You already heave `opt_value` variable above? Is there a reason to redefine it here?

Reply:
Oops, that's a mistake, missed that while refactoring...